scalafix-core/shared/src/main/scala/scalafix/v1/Sym.scala

```diff
@@ -0,0 +1,149 @@
+package scalafix.v1
+
+import scala.meta.Tree
+import scala.meta.internal.{semanticdb3 => s}
+import scala.meta.internal.semanticdb3.SymbolInformation.{Property => p}
+import scala.meta.internal.semanticdb3.Scala._
+
+final class Sym private (val value: String) {
+  def isNone: Boolean = value.isNone
+  def isRootPackage: Boolean = value.isRootPackage
+  def isEmptyPackage: Boolean = value.isEmptyPackage
+  def isGlobal: Boolean = value.isGlobal
+  def isLocal: Boolean = value.isLocal
+  def owner: Sym = Sym(value.owner)
+  def info(doc: SemanticDoc): Sym.Info = doc.info(this)
+
+  override def toString: String =
+    if (isNone) "Sym.None"
+    else value
+  override def equals(obj: Any): Boolean =
+    this.eq(obj.asInstanceOf[AnyRef]) || (obj match {
+      case s: Sym => value == s.value
+      case _ => false
+    })
+  override def hashCode(): Int = value.hashCode
+}
+
+object Sym {
+  val RootPackage: Sym = new Sym(Symbols.RootPackage)
+  val EmptyPackage: Sym = new Sym(Symbols.EmptyPackage)
+  val None: Sym = new Sym(Symbols.None)
+  def apply(sym: String): Sym = {
+    if (sym.isEmpty) Sym.None
+    else {
+      if (!sym.startsWith("local")) {
+        sym.desc // assert that it parses as a symbol
+      }
+      new Sym(sym)
+    }
+  }
+
+  object Root {
+    def unapply(sym: Sym): Boolean =
+      sym.isRootPackage
+  }
+
+  object Local {
+    def unapply(sym: Sym): Option[Sym] =
+      if (sym.isLocal) Some(sym) else scala.None
+  }
+
+  object Global {
+    def unapply(sym: Sym): Option[(Sym, Sym)] =
+      if (sym.isGlobal) {
+        val owner = Sym(sym.value.owner)
+        Some(owner -> sym)
+      } else {
+        scala.None
+      }
+  }
+
+  final class Info private[scalafix] (
+      private[scalafix] val info: s.SymbolInformation
+  ) {
+    def isNone: Boolean = info.symbol.isEmpty
+    def sym: Sym = new Sym(info.symbol)
+    def owner: Sym = new Sym(info.owner)
+    def name: String = info.name
+    def kind: Kind = new Kind(info)
+    def props: Properties = new Properties(info.properties)
+    def access: Accessibility =
+      new Accessibility(info.accessibility.getOrElse(s.Accessibility()))
+
+    override def toString: String = s"Sym.Info(${info.symbol})"
+
+    // privates
+    private[scalafix] def tpe: s.Type =
+      info.tpe.getOrElse(s.Type())
+  }
+  object Info {
+    val empty = new Sym.Info(s.SymbolInformation())
+  }
+
+  final class Kind private[Sym] (info: s.SymbolInformation) {
+    def isClass: Boolean = k.isClass
+    def isObject: Boolean = k.isObject
+    def isTrait: Boolean = k.isTrait
+    def isMethod: Boolean = k.isMethod
+    def isField: Boolean = k.isField
+    def isMacro: Boolean = k.isMacro
+    def isConstructor: Boolean = k.isConstructor
+    def isType: Boolean = k.isType
+    def isParameter: Boolean = k.isParameter
+    def isTypeParameter: Boolean = k.isTypeParameter
+    def isPackage: Boolean = k.isPackage
+    def isPackageObject: Boolean = k.isPackageObject
```

Comment:
Looks like there are a few SemanticDB kinds missing from this API. Is this intentional?

Reply:
Good catch, updated to match 100% with semanticdb3.proto (including order).

Only missing Kind.LOCAL which is going away.